wandb/sdk/wandb_setup.py

```diff
@@ -161,32 +174,31 @@ def _get_logger(self):
     def settings(self):
         return self._settings
 
-    def _clone_settings(self, __d=None, **kwargs):
-        s = copy.copy(self._settings)
-        s.update(__d, **kwargs)
-        return s
-
-    def _get_entity(self):
+    def _get_entity(self) -> Optional[str]:
         if self._settings and self._settings._offline:
             return None
         if self._server is None:
             self._load_viewer()
         entity = self._server._viewer.get("entity")
         return entity
 
-    def _load_viewer(self, settings=None):
+    def _load_viewer(self, settings=None) -> None:
         if self._settings and self._settings._offline:
             return
         s = server.Server(settings=settings)
         s.query_with_timeout()
         self._server = s
 
-    def _load_user_settings(self, settings=None):
+    def _load_user_settings(self, settings=None) -> Optional[Dict[str, Any]]:
         if self._server is None:
-            self._load_viewer()
+            self._load_viewer(settings=settings)
+
+        # offline?
+        if self._server is None:
+            return None
```

Comment:
Can be under the same condition. Why do you need to change this? How did it use to work?

Reply:
ah, this came up in `https://wandb.atlassian.net/browse/WB-7410` and is actually what this whole refactoring thing originated from! the user's environment was messed up, which should not be an issue anymore with the new settings (things would either be ok or error out earlier + since now run.settings is exposed, it's much easier to debug). 
why needing two if clauses: if _server is None, try setting it up, and if that that still fails, don't do anything. the hope is that this will now never happen :)

Reply:
Ah I see now... makes sense... should we leave a comment here?